### PR TITLE
Implement a plugin architecture into flight-config

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
Previously `flight-config` required all the configs to be defined within a single `flight-config-map.yml` file. This makes it difficult for external applications to integrate into `flight-config`.

Instead external apps can drop in there config definitions into `flght-config-map.d` directory. These additional configs are deep merged over the top of the original config. This means each "category" is not exclusively owned by a single config definition.

In addition to the above changes, the `rc` and `cshrc` files no longer need to exist before defining the config. The `set` command now implicitly creates the `rc` files.

### CODE REVIEW

**Comment by benarmston** [comment]
@WilliamMcCumstie completely contradicting what I said yesterday, I can't see any reason to have `/opt/flight/etc/flight-config-map.yml` in addition to `/opt/flight/etc/flight-config-map.d/`.  I think the code would be simpler if we moved `/opt/flight/etc/flight-config-map.yml` to, say, `/opt/flight/etc/flight-config-map.d/cluster.yml` and dropped support for `/opt/flight/etc/flight-config-map.yml`.

File: dist/opt/flight/etc/profile.d/90-locale-check.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Starter, please visit:
+# https://github.com/openflighthpc/flight-starter
+#==============================================================================
+
+function warn_locale() {
+  cat <<EOF >&2
+# Your locale appears to be set inconsistently or does not use unicode.
+# This could cause various application to break unexpectedly.
```
Comment (benarmston) [inline dist/opt/flight/etc/profile.d/90-locale-check.sh:31]: *applications*

File: dist/opt/flight/etc/profile.d/90-locale-check.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Starter, please visit:
+# https://github.com/openflighthpc/flight-starter
+#==============================================================================
+
+function warn_locale() {
+  cat <<EOF >&2
+# Your locale appears to be set inconsistently or does not use unicode.
```
Comment (benarmston) [inline dist/opt/flight/etc/profile.d/90-locale-check.sh:30]: I think `unicode` should be capitalized.  At least that's how https://home.unicode.org/ and wikipedia treat it.

File: dist/opt/flight/etc/profile.d/90-locale-check.csh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# For more information on Flight Starter, please visit:
+# https://github.com/openflighthpc/flight-starter
+#==============================================================================
+
+if ( ! $?flight_ROOT ) then
+  setenv flight_ROOT /opt/flight
+endif
+
+setenv flight_LOCALE_cmds "setenv LANG en_US.utf8\nsetenv LC_ALL en_US.utf8"
+bash --norc --noprofile $flight_ROOT/etc/profile.d/90-locale-check.sh
```
Comment (benarmston) [inline dist/opt/flight/etc/profile.d/90-locale-check.csh:33]: Nice way to avoid scripting in `tcsh` :-)

**Review by benarmston — COMMENTED** [review]
A couple of copy changes for you to look at please.

File: dist/opt/flight/etc/profile.d/90-locale-check.csh (hunk 3 above)
Comment (WilliamMcCumstie) [reply dist/opt/flight/etc/profile.d/90-locale-check.csh:33]: Life goals

File: dist/opt/flight/etc/profile.d/90-locale-check.sh (hunk 2 above)
Comment (WilliamMcCumstie) [reply dist/opt/flight/etc/profile.d/90-locale-check.sh]: https://github.com/openflighthpc/flight-starter/pull/5/commits/877117adf31725e9266360e4860eb4605f29e040

File: dist/opt/flight/etc/profile.d/90-locale-check.sh (hunk 1 above)
Comment (WilliamMcCumstie) [reply dist/opt/flight/etc/profile.d/90-locale-check.sh]: https://github.com/openflighthpc/flight-starter/pull/5/commits/877117adf31725e9266360e4860eb4605f29e040

**Comment by benarmston** [comment]
This is good to go.

**Pull request merged**